pkg/resource/asset.go

```diff
@@ -38,7 +38,12 @@ import (
 
 	"github.com/pulumi/pulumi/pkg/util/contract"
 	"github.com/pulumi/pulumi/pkg/util/httputil"
-	"github.com/pulumi/pulumi/pkg/workspace"
+)
+
+const (
+	// BookkeepingDir is the name of our bookkeeping folder, we store state here (like .git for git).
+	// Copied from workspace.BookkeepingDir to break import cycle.
```

Comment:
What caused the import cycle?

Reply:
Adding:

```
import (
	"github.com/pulumi/pulumi/pkg/resource"
)
```

to `pkg/resource/config/map.go` (to make use of `ParsePropertyPath`) introduces the import cycle:

```
import cycle not allowed
package github.com/pulumi/pulumi
	imports github.com/pulumi/pulumi/cmd
	imports github.com/pulumi/pulumi/pkg/apitype
	imports github.com/pulumi/pulumi/pkg/resource
	imports github.com/pulumi/pulumi/pkg/workspace
	imports github.com/pulumi/pulumi/pkg/resource/config
	imports github.com/pulumi/pulumi/pkg/resource
```

Because `github.com/pulumi/pulumi/pkg/resource` imports `workspace` (for the bookkeeping dir) and `workspace` imports `github.com/pulumi/pulumi/pkg/resource/config` for some config stuff.